libsolidity/analysis/DeclarationTypeChecker.cpp

```diff
@@ -456,8 +457,18 @@ void DeclarationTypeChecker::endVisit(VariableDeclaration const& _variable)
 	}
 	else if (_variable.isStateVariable())
 	{
-		solAssert(varLoc == Location::Unspecified, "");
-		typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
+		switch (varLoc)
+		{
+			case Location::Unspecified:
+				typeLoc = (_variable.isConstant() || _variable.immutable()) ? DataLocation::Memory : DataLocation::Storage;
+				break;
+			case Location::Transient:
+				typeLoc = DataLocation::Transient;
```

Comment:
I think you should assert that we're not dealing with a constant or an immutable. `immutable transient` or `transient constant` should be rejected by the parser and never reach this place.

Reply:
Done. Now parser rejects `transient` with immutable or constant.